frame/evm/src/runner/stack.rs

```diff
@@ -1170,6 +1183,13 @@ where
 			code.len(),
 			address
 		);
+
+		// The EVM returns empty code in case of an EIP-7702 delegation to the zero address.
+		// In response we must remove such delegation from the authorizing account.
+		if code.is_empty() {
+			Pallet::<T>::remove_account_code(&address);
+		}
+
 		Pallet::<T>::create_account(address, code, caller)
```

Comment:
Another problem that I see is that we call `frame_system::Pallet::<T>::inc_sufficients(who);` on every call to `Pallet::<T>::create_account(address, code, caller)`, which usually uses `FrameSystemAccountProvider`. This is not ideal, since it will be possible to call `inc_sufficients` many times.

In my opinion, we should update this method signature with a new argument `from_authorization: bool`. Which when `true`, the call to `Pallet::<T>::create_account(address, code, caller)` would be skipped.

```rs
fn set_code(
	&mut self,
	address: H160,
	code: Vec<u8>,
	caller: Option<H160>,
	from_authorization: bool,
)
```

Or add a new function to the trait `fn set_authorization_code` or `set_eoa_code`, which would be much cleaner and simpler.

Reply:
Opened a PR on `evm` crate: https://github.com/rust-ethereum/evm/pull/367